src/Symfony/Component/OptionsResolver/Debug/OptionsResolverIntrospector.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\OptionsResolver\Debug;
+
+use Symfony\Component\OptionsResolver\Exception\NoConfigurationException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ *
+ * @final
+ */
+class OptionsResolverIntrospector
+{
+    private $optionsResolver;
+    private $get;
+
+    public function __construct(OptionsResolver $optionsResolver)
+    {
+        $this->optionsResolver = $optionsResolver;
+
+        $this->get = \Closure::bind(function ($property, $option, $message) {
+            if (!array_key_exists($option, $this->{$property})) {
+                throw new NoConfigurationException($message);
+            }
+
+            return $this->{$property}[$option];
+        }, $optionsResolver, $optionsResolver);
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return mixed
+     *
+     * @throws NoConfigurationException on no configured value
+     */
+    public function getDefault($option)
+    {
+        return call_user_func($this->get, 'defaults', $option, sprintf('No default value was set for the "%s" option', $option));
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return callable[]
+     *
+     * @throws NoConfigurationException on no configured closures
+     */
+    public function getLazyClosures($option)
+    {
+        return call_user_func($this->get, 'lazy', $option, sprintf('No lazy closures were set for the "%s" option', $option));
+    }
+
+    /**
+     * @param string $option
+     *
+     * @return string[]
```

Comment:
Not exactly true, `\Closure` are allowed too, so `mixed[]`?

Reply:
`\Closure` are only allowed for allowed values, not allowed types, right?

Reply:
Right, I confused it with `allowedValues` ;)